src/backend/joanie/core/models/products.py

```diff
@@ -880,6 +892,25 @@ def save(self, *args, **kwargs):
         self.full_clean()
         super().save(*args, **kwargs)
 
+    def get_total_price(self):
+        """Return the total price considering the order group discount."""
+        if (
+            self.order_group
+            and self.order_group.is_enabled
+            and self.order_group.discount
+        ):
+            discount = self.order_group.discount
+            price = self.product.price
+
+            if discount.rate:
+                total = price - (price * Decimal(str(discount.rate)))
+                return total.quantize(TOTAL_DECIMAL_PLACES_ACCEPTED)
+
+            if discount.amount:
+                return price - discount.amount
+
+        return self.product.price
```

Comment:
The return early pattern seems to fit well in this case
```suggestion
        if (
            not self.order_group
            or not self.order_group.is_enabled
            or not self.order_group.discount
        ):
            return self.product.price

        discount = self.order_group.discount
        price = self.product.price

        if discount.rate:
            total = price - (price * Decimal(str(discount.rate)))
            return total.quantize(TOTAL_DECIMAL_PLACES_ACCEPTED)

        if discount.amount:
            return price - discount.amount
```